src/screens/GroupSetting/hooks/useGroupSetting.js

```diff
@@ -36,53 +49,73 @@ const useGroupSetting = ({navigation, route}) => {
 
-  const submitData = async (withNavigation = true, withLoading = true) => {
-    let changeImageUrl = '';
-    if (changeImage) {
-      if (withLoading) setIsLoading(true);
-      try {
-        const res = await uploadFile(`data:image/jpeg;base64,${base64Profile}`);
-        changeImageUrl = res.data.url;
-      } catch (e) {
-        setIsLoading(false);
-        return;
-      }
+  React.useEffect(() => {
+    if (selectedChannel) {
+      setUrlImage(selectedChannel?.channelPicture);
     }
+  }, [selectedChannel]);
 
-    if (changeName || changeImage) {
-      if (withLoading) setIsLoading(true);
-      let dataEdit = {
-        name: groupName
-        // ...(changeImage && {image: base64Profile}),
+  const submitData = async (withNavigation = true) => {
+    if (!changeImage || changeName) return navigation.goBack();
+    try {
+      setLoadingUpdate(true);
+      let body = {
+        channel_id: route?.params?.id,
+        channel_name: groupName
       };
-      if (changeName) {
-        dataEdit = {...dataEdit, isEditName: true};
+      if (urlImage) {
+        try {
+          const res = await uploadPhotoImage(urlImage);
+          console.log({res}, 'nana');
+          body = {...body, channel_image: res.data.url};
+        } catch (e) {
+          console.log(e, 'error upload');
+        }
       }
-      if (changeImage) {
-        dataEdit.image = changeImageUrl;
-      } else if (channel?.data?.image) {
-        dataEdit.image = channel?.data?.image;
+      const response = await SignedMessageRepo.editChannel(body);
+      let newChannel = response.data?.channel;
+      newChannel = {
+        ...chat.selectedChannel,
+        ...newChannel,
+        channelType: handleChannelType(newChannel?.channelType),
+        channelPicture: body?.channel_image
+      };
+      ChannelList.updateChannelInfo(
+        localDb,
+        body.channel_id,
+        body.channel_name,
+        body.channel_image
+      );
+      if (newChannel) {
+        setChat({
+          ...chat,
+          selectedChannel: newChannel
+        });
       }
-      console.log({dataEdit}, 'nusu');
-      updateDataEdit(dataEdit, withNavigation);
-    } else if (withNavigation) navigation.goBack();
-  };
+      refresh('channelInfo');
+      refresh('chat');
 
-  const updateDataEdit = async (dataEdit, withNavigation) => {
-    try {
-      await channel.update(dataEdit);
-      if (withNavigation) navigation.navigate('ChatDetailPage');
-    } catch (e) {
-      if (__DEV__) {
-        console.log(`error : ${e}`);
+      refresh('channelList');
+
+      if (withNavigation) {
+        navigation.goBack();
       }
+      setLoadingUpdate(false);
+    } catch (e) {
+      setLoadingUpdate(false);
       SimpleToast.show(StringConstant.groupSettingUpdateFailed);
     }
-    setIsLoading(false);
+  };
+
+  const handleChannelType = (channelType) => {
+    if (channelType === 1) {
+      return CHANNEL_GROUP;
+    }
+    return channelType;
   };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `submitData` function has been significantly updated. The new code now handles image uploads and channel editing. It also uses the `SignedMessageRepo.editChannel` method to send the edited data to the server. The error handling in this function could be improved by providing more specific error messages based on the type of error caught.

```diff
- const submitData = async (withNavigation = true, withLoading = true) => {
-   let changeImageUrl = '';
-   if (changeImage) {
-     if (withLoading) setIsLoading(true);
-     try {
-       const res = await uploadFile(`data:image/jpeg;base64,${base64Profile}`);
-       changeImageUrl = res.data.url;
-     } catch (e) {
-       setIsLoading(false);
-       return;
-     }
-    }
- 
-   if (changeName || changeImage) {
-     if (withLoading) setIsLoading(true);
-     let dataEdit = {
-       name: groupName
-       // ...(changeImage && {image: base64Profile}),
-      };
-     if (changeName) {
-       dataEdit = {...dataEdit, isEditName: true};
-      }
-     if (changeImage) {
-       dataEdit.image = changeImageUrl;
-     } else if (channel?.data?.image) {
-       dataEdit.image = channel?.data?.image;
-      }
-     console.log({dataEdit}, 'nusu');
-     updateDataEdit(dataEdit, withNavigation);
-   } else if (withNavigation) navigation.goBack();
- };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->